storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -5006,6 +5006,42 @@ static bool xb_space_create_file(
   return (true);
 }
 
+/************************************************************************
+get space_id from page 0
+@return space_id or SPACE_UNKOWN */
+static space_id_t get_space_id_from_page_0(
+    const char *file_name /*!<in: file name */) {
+  bool ok;
+  space_id_t space_id = SPACE_UNKNOWN;
+
+  auto buf = static_cast<byte *>(ut_malloc_nokey(2 * srv_page_size));
+
+  auto file = os_file_create_simple_no_error_handling(
+      0, file_name, OS_FILE_OPEN, OS_FILE_READ_ONLY, srv_read_only_mode, &ok);
+
+  if (ok) {
+    auto *page = static_cast<buf_frame_t *>(ut_align(buf, srv_page_size));
+
+    IORequest request(IORequest::READ);
+    dberr_t err = os_file_read_first_page(
+        request, dict_sys_t::s_dd_space_file_name, file, page, UNIV_PAGE_SIZE);
```

Comment:
This proves filename is irrelevant once file is opened.. Only used for printing filename in error messages